SU2_CFD/src/solvers/CScalarSolver.cpp

```diff
@@ -450,12 +448,17 @@ void CScalarSolver::SetResidual_DualTime(CGeometry* geometry, CSolver** solver_c
   unsigned short iVar, iMarker, iDim, iNeigh;
   unsigned long iPoint, jPoint, iVertex, iEdge;
 
-  const su2double *U_time_nM1 = nullptr, *U_time_n = nullptr, *U_time_nP1 = nullptr;
+  su2double *U_time_nM1 = nullptr, *U_time_n = nullptr, *U_time_nP1 = nullptr;
+  su2double Density_nM1 = 1.0, Density_n = 1.0, Density_nP1 = 1.0;
   su2double Volume_nM1, Volume_nP1;
-  su2double Density_nM1, Density_n, Density_nP1;
   const su2double *Normal = nullptr, *GridVel_i = nullptr, *GridVel_j = nullptr;
   su2double Residual_GCL;
 
+  const int nvar = nVar;
+  auto FactorTimesVector = [nvar](su2double factor, su2double* vec) {
+    for (int iVar = 0; iVar < nvar; iVar++) vec[iVar] = factor * vec[iVar];
+  };
```

Comment:
Maybe this is already implemented somewhere centrally? (Note that I wanted to avoid the term Scalar ...)

Reply:
nvm, removed that